operator/bucketcontroller/create.go

```diff
@@ -54,6 +57,27 @@ func (p *ProvisioningPipeline) createS3Bucket(ctx *pipelineContext) error {
 	return nil
 }
 
+// enrichConnectionSecret adds more data to the already existing secret such as EndpointName, RegionName and BucketName
+func (p *ProvisioningPipeline) enrichConnectionSecret(ctx *pipelineContext) error {
+	bucket := ctx.bucket
+	log := controllerruntime.LoggerFrom(ctx)
+
+	secretRef := bucket.Spec.ForProvider.CredentialsSecretRef
```

Comment:
I disagree here that we enrich the connection secret that has the credentials for making buckets.
I would create a completely new Secret that has the credentials, but also the endpoint and region. That way we don't interfere with the objectsuser controller, which is and should be the sole owner of the secret.
```suggestion
	secretRef := bucket.Spec.WriteConnectionSecretToReference
```

Reply:
I would really avoid creating another secret, it's already very confusing how crossplane is managing secrets. Adding another secret on top would unnecessarily add complexity which we need to reduce were possible.

Reply:
IMO this is practically not even up for debate. I mean, consider the following case:

We can create multiple buckets (for multiple regions) from the same objectuser. How do you handle different buckets in different regions? Both endpoint and region change, so the secret needs to contain a mapping which endpoint and region belongs to which bucket while all the buckets share the same credentials.

Sure, currently in VSHN's AppCat offering we maintain a 1:1 relationship in a composition. But, firstly the provider controllers don't know that these resources are managed by a higher abstraction layer (and shouldn't). Second this relationship can change at composition level and we wouldn't want to introduce breaking changes in secret handling in the provider just because VSHN's offering changes.

Reply:
Well the update was specifically for `1:1 `relationship, but it's definitely not good for `1:n`. Then let's anticipate that and create another secret.

Reply:
Updated the code and now we are creating a new secret for bucket that has all the info necessary.